include/libdnf/rpm/package_query.hpp

```diff
@@ -605,6 +605,7 @@ class PackageQuery : public PackageSet {
     friend libdnf::Goal;
     class Impl;
     Flags flags;
+    std::unique_ptr<libdnf::solv::SolvMap> considered_cached;
```

Comment:
Hmm `libdnf::solv::SolvMap` is an incomplete type. "package_query.hpp" is a public header and `libdnf::solv::SolvMap` is an internal type of libdnf. But `std::optional` requires full type. That means a bigger change. Add another instance of impl. One impl is in `PackageSet`. And `PackageQuery` inherits from `PackageSet`.

Reply:
So I'll add an implementation class.

Reply:
I see.. couldn't you use `PackageSet` instead of `SolvMap` here then? The public `PackageSet` wrapper to the private `SolvMap` is causing some internal schizophrenia :slightly_smiling_face: 

Or what's your solution? I'd prefer not to add more classes...